semantics/language/execution/stmt/return.k

```diff
@@ -51,8 +52,8 @@ module C-STMT-RETURN
                     | "Return''" "(" K ")"
      context Return'(HOLE => reval(HOLE)) [result(RValue)]
 
-     /* recursively exist each Block wrapped in BlockStatement */
-     rule <k> (.K => exitRestrictScope(BlockNum) ~> deleteLocals ~> popBlock)
+     /* recursively exits each Block wrapped in BlockStatement */
+     rule <k> (.K => exitRestrictScope(BlockNum) ~> popBlock)
```

Comment:
I did refactoring all over the place. If this re-introduces Andrei's issue (it'd be nice if there was a regression test, of course) we can re-fix it, but how blocks were dealt with was all refactored (Block gets expanded to pushBlock ~> ... ~> popBlock during translation and popBlock => deleteLocals).